### DISCUSSION TITLE
Is it safe to host a ruby on rails app publicly in development mode?

### DISCUSSION THREAD

**Discussion by danielkinahan** [Freika/dawarich#101] [General]
I've never worked with this framework before, only Django, which doesn't recommend this. I'd like to expose this publicly to connect to Overland but I'm unsure if that's a good idea.

**Comment by Freika** [comment]
Hi @danielkinahan! At this point, I'd advise against exposing Dawarich publicly, although I already know of the cases of such exposure.

Instead of publicly exposing it, I'd rather suggest using something like https://tailscale.com/, it will take like 5 minutes for you to have access from your mobile device (or any device for that matter) to your home server.

**Reply by danielkinahan** [reply]
Hey thanks for the quick reply. I'm not interested in tying my home server to 3rd party infrastructure but I am curious about what the reason behind leaving this project in development mode is. What is required for this project to transition to production?

**Reply by Freika** [reply]
Since the project is being actively developed, it's crucial to show as much debugging info to users as possible so they could report problems with as much context as possible.

If you have any experience with Ruby on Rails, or want to learn, you could fork the project and prepare it for production, using, say, official guides on configuration: https://guides.rubyonrails.org/configuring.html

**Reply by samtate** [reply]
@danielkinahan if you aren't interested in third party infrastructure, take a look at headscale, which is a self hosted version of Tailscale. Otherwise, look at setting up a simple Wireguard VPN
